pyaerocom/io/read_eea_aqerep_base.py

```diff
@@ -563,7 +583,6 @@ def read(
             if len(self.files) == 0:
                 logger.info("Retrieving file list")
                 files = self.get_file_list(self.FILE_MASKS[var_name])
```

Comment:
I can not see where is `files` defined, and what will happen when `self.files` is not empty

Reply:
> files is defined here:

I see, `files` is one of the method arguments. I totally missed it

> For EEA reading self.files should always be empty from now on. The OS caching should work quite well here. I have removed the check for self.files now.

If you are assuming `self.files` to be empty and `files` to be None,
I think you should put an assert at the top of the method to make the assumption clear.
Even better would be to put a guarding clause that raises a meaningful exception

``` python
    def read(
        self, vars_to_retrieve=None, files=None, first_file=None, last_file=None, metadatafile=None
    ):
    [...]
    if self.files:
        raise RuntimeError(f"{self.files=} should be empty")
    if files is not None:
        raise ValueError(f"{files=} should be None")

    logger.info("Retrieving file list")
    files = self.get_file_list(self.FILE_MASKS[var_name])
    [...]  
```

Reply:
`files` can be not None in the case the user wants supply a list of files to be read. All other readers have the same (e.g. Aeronet:
https://github.com/metno/pyaerocom/blob/367d5cb4e68e2a4557ccc6da09bece6495a084f0/pyaerocom/io/readaeronetbase.py#L298-L306
 Very useful for testing. Therefore I would suggest to keep that functionality.

Searching through the code shows that all other readers (except for the ones for Aeronet) actually fill `self.files`. I therefore think, the `RuntimeError` should not be thrown either.

We should unify the obs readers...

Reply:
> Very useful for testing. Therefore I would suggest to keep that functionality.

Fair point.

> Searching through the code shows that all other readers (except for the ones for Aeronet) actually fill `self.files`

In that case, this reader should do the same.

Reply:
I added the assignment to `self.files` again. It's just not used for caching anymore.